enhancements/rhcos/rhcos-cloud-marketplace-images.md

```diff
@@ -0,0 +1,330 @@
+x64             rh-ocp-worker  redhat-rhel     rh-ocp-worker       redhat-rhel:rh-ocp-worker:rh-ocp-worker:4.13.2023101100            4.13.2023101100
+x64             rh-ocp-worker  redhat-limited  rh-ocp-worker-gen1  redhat-limited:rh-ocp-worker:rh-ocp-worker-gen1:4.8.2021122100     4.8.2021122100
+x64             rh-ocp-worker  redhat-limited  rh-ocp-worker-gen1  redhat-limited:rh-ocp-worker:rh-ocp-worker-gen1:413.92.2023101700  413.92.2023101700
+x64             rh-ocp-worker  redhat-rhel     rh-ocp-worker-gen1  redhat-rhel:rh-ocp-worker:rh-ocp-worker-gen1:4.13.2023101100       4.13.2023101100
+```
+
+Images from the `redhat-limited` publisher are for EMEA while `RedHat` is for North America. Free marketplace images could be published under
+a new offer, such as: `RHCOS`, `OpenShift` or, keeping in line with the current convention, `rh-ocp`, such that the listing would look like:
+
+```shell
+$  az vm image list --all --offer rhcos --publisher redhat -o table
+Architecture    Offer          Publisher       Sku                 Urn                                          Version
+--------------  -------------  --------------  ------------------  -------------------------------------------  -----------------
+x64             rhcos          RedHat          rhcos               RedHat:rhcos:rhcos:417.94.202407010929       417.94.202407010929
+x64             rhcos          RedHat          rhcos-gen1          RedHat:rhcos:rhcos-gen1:417.94.202407010929  417.94.202407010929
+```
+
+##### Azure Stream Representation
+
+An Azure Marketplace extension, `azure-marketplace`, would be added, with unpaid and paid (both North America & EMEA)
+marketplace images listed as child objects: 
+
+```yaml
+"rhel-coreos-extensions": {
+  "azure-disk": {
+    "release": "417.94.202407010929-0",
+    "url": "https://rhcos.blob.core.windows.net/imagebucket/rhcos-417.94.202407010929-0-azure.x86_64.vhd"
+  },
+  "azure-marketplace": {
+    "no-purchase-plan": {
+      "publisher": "RedHat",
+      "offer": "rhcos",
+      "sku": "rhcos",
+      "version": "417.94.202407010929"
+    },
+    "purchase-plan-north-america": {
+      "publisher": "RedHat",
+      "offer": "rh-ocp-worker",
+      "sku": "rh-ocp-worker",
+      "version": "417.94.202407010929"
+    },
+    "purchase-plan-emea": {
+      "publisher": "redhat-limited",
+      "offer": "rh-ocp-worker",
+      "sku": "rh-ocp-worker",
+      "version": "417.94.202407010929"
+    }
+  }
+```
+
+[Hyperv Generation 1 images](https://learn.microsoft.com/en-us/azure/virtual-machines/generation-2#features-and-capabilities)
+would be published, identified by a `gen1` suffix, but would not be included in the RHCOS stream. Any time that the installer
+or users need to select a gen1 image, they would simply add the `gen1` suffix to the SKU identified in the RHCOS stream.  
+
+OPEN QUESTION: How to handle EMEA vs North America 
+OPEN QUESTION: how to handle azure gen1 image
+
```

Comment:
Should another open question be how we handle multi-cloud images in Azure?  Is govcloud in scope?  China cloud?  etc.

Reply:
One more thing is how we as an organization should ask the release engineering / spstrat team (assuming they're the ones owning the upload process) how to structure the images, specifically in Azure.

Azure is broken down as such:
- publisher: consistent
- offer: can make new offers, but typically you'd split fcos and rhcos here
- version: the version of the rhcos vhd
- sku: this is where things get interesting.  Marketplace allows configuring different options the OS supports such as trusted launch, cloud-init, supports sriov, etc.  Typically openshift minor releases will enable new functionality such as this, hence why it would be good to configure a new SKU for every minor version so we can support / enable these features for customers.   


When configuring images in the marketplace, you configure them at a plan level

Reply:
> Should another open question be how we handle multi-cloud images in Azure? Is govcloud in scope? China cloud? etc.

I think it's a good question, and one we will need to answer for the implementation, but I don't think it affects the overall enhancement. I'm happy to add it as an open question in the next update.

Reply:
> sku: this is where things get interesting. Marketplace allows configuring different options the OS supports such as trusted launch, cloud-init, supports sriov, etc. Typically openshift minor releases will enable new functionality such as this, hence why it would be good to configure a new SKU for every minor version so we can support / enable these features for customers.

I have incorporated this into the enhancement. The sku will be variable, but deterministic.